gcp_variant_transforms/libs/header_merger.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2019 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""class for merging 2 VcfHeader objects"""
+
+from collections import OrderedDict
+from typing import Dict, Any  #pylint: disable=unused-import
+
+from gcp_variant_transforms.beam_io import vcf_header_io
+from gcp_variant_transforms.libs import vcf_field_conflict_resolver #pylint: disable=unused-import
+
+class HeaderMerger(object):
+  """Class for merging two :class:`VcfHeader`s."""
+
+  def __init__(self, resolver):
+    # type: (vcf_field_conflict_resolver.FieldConflictResolver) -> None
+    """Initialize :class:`VcfHeader` object.
+
+    Args:
+      resolver: Auxiliary class for resolving possible header value mismatches.
+    """
+    self._resolver = resolver
+
+  def merge(self, first, second):
+    # type: (vcf_header_io.VcfHeader, vcf_header_io.VcfHeader) -> None
+    """Updates `first`'s headers with values from `second`.
+
+    If a specific key does not already exist in a specific one of `first`'s
+    headers, that key and the associated value will be added. If the key does
+    already exist in the specific header of `first`, then the value of that
+    key will be updated with the value from `second`.
+
+    Args:
+      first: The VcfHeader object.
+      second: The VcfHeader object that's headers will be merged into the
+        headers of first.
+    """
+    if (not isinstance(first, vcf_header_io.VcfHeader) or
+        not isinstance(first, vcf_header_io.VcfHeader)):
```

Comment:
Thanks for catching the issues for merge header definitions. s/first/second.

Reply:
Oops, I didn't realize it happened in this file as well :D .
Done.